### PR TITLE
enhancement(notification): add white list check for webhook to prevent SSRF

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-enhancement

#### What this PR does / why we need it:

The  webhook url was not validated, this may cause SSRF attack.
So we added a system configuration `odc.notification.host-white-list` , only allow to create channel with webhook in this list.

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/ChannelConfigValidator.java [hunk 1]
```diff
@@ -103,4 +116,16 @@ private void validateWebhookChannelConfig(WebhookChannelConfig channelConfig) {
 
     }
 
+    private boolean checkHostInWhiteList(String webhook) {
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/ChannelConfigValidator.java:119]: ODC already existed an SSRF Checker, reference to `PreConditions#validInHostWhiteList`

Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/ChannelConfigValidator.java]: > seems we can use PreConditions.validInUrlWhiteList straightly. which is configured by system configuration `odc.integration.url-white-list straightly`

Should we not mix the white lists of these two modules?

Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/ChannelConfigValidator.java]: > > seems we can use PreConditions.validInUrlWhiteList straightly. which is configured by system configuration `odc.integration.url-white-list straightly`
> 
> Should we separate the white lists of these two modules?

I'm okay ,as it is security part, not business feature

Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/ChannelConfigValidator.java]: sure, I've fixed this.
Now we use the `SSRFChecker` straightly and the config `odc.integration.url-white-list `

**Review by yizhouxw — APPROVED** [review]

**Review by ungreat — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by smallsheeeep — APPROVED** [review]
lgtm

**Pull request merged**